src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/NamespaceCache.java

```diff
@@ -72,49 +84,64 @@ protected Uni<CollectionSettings> getCollectionProperties(
                   if ((error instanceof StatusRuntimeException sre
                       && (sre.getStatus().getCode() == io.grpc.Status.Code.NOT_FOUND
                           || sre.getStatus().getCode() == io.grpc.Status.Code.INVALID_ARGUMENT))) {
                     return Uni.createFrom()
                         .item(
-                            new CollectionSettings(
+                            new CollectionSchemaObject(
+                                namespace,
                                 collectionName,
-                                CollectionSettings.IdConfig.defaultIdConfig(),
-                                CollectionSettings.VectorConfig.notEnabledVectorConfig(),
+                                CollectionSchemaObject.IdConfig.defaultIdConfig(),
+                                VectorConfig.notEnabledVectorConfig(),
                                 null));
                   }
                   return Uni.createFrom().failure(error);
                 } else {
-                  vectorCache.put(collectionName, result);
+                  schemaObjectCache.put(collectionName, result);
                   return Uni.createFrom().item(result);
                 }
               });
     }
   }
 
-  private Uni<CollectionSettings> getVectorProperties(
+  private Uni<SchemaObject> loadSchemaObject(
       DataApiRequestInfo dataApiRequestInfo, String collectionName) {
+
     return queryExecutor
         .getSchema(dataApiRequestInfo, namespace, collectionName)
         .onItem()
         .transform(
-            table -> {
-              if (table.isPresent()) {
-                // check if its a valid json api table
-                if (!new JsonapiTableMatcher().test(table.get())) {
-                  throw new JsonApiException(
-                      ErrorCode.INVALID_JSONAPI_COLLECTION_SCHEMA,
-                      ErrorCode.INVALID_JSONAPI_COLLECTION_SCHEMA.getMessage() + collectionName);
-                }
-                return CollectionSettings.getCollectionSettings(table.get(), objectMapper);
-              } else {
-                throw new RuntimeException(
-                    ErrorCode.COLLECTION_NOT_EXIST.getMessage() + collectionName);
+            optionalTable -> {
+              // TODO: AARON - I changed the logic here, needs to be checked
+              // TODO: error code here needs to be for collections and tables
+              var table =
+                  optionalTable.orElseThrow(
+                      () ->
+                          new RuntimeException(
+                              ErrorCode.COLLECTION_NOT_EXIST.getMessage() + collectionName));
+
+              // check if its a valid json api table
+              // TODO: re-use the table matcher this is on the request hot path
+              if (new JsonapiTableMatcher().test(table)) {
+                return CollectionSchemaObject.getCollectionSettings(
+                    optionalTable.get(), objectMapper);
+              }
+
```

Comment:
Just to be clear, so if it is not a collection, must be a table.

Reply:
Yes, I think that is the assumption (and logic) here.